### PR TITLE
refactor: simplify controller for getting element HTML

### PR DESCRIPTION
**Pull request by andreasphil**
RISDEV-3899

### CODE REVIEW

**Comment by andreasphil** [comment]
@hannes-mk I realized we could leave the date parsing to Spring by using `Instant` as the parameter type (rather than `String` + parse that ourselves). If we don't need to attempt the parsing ourselves we can simplify the controller a lot. Only downside is that we can't have a 400 status code then. I think that's fine but would be curious to hear your feedback.

**Comment by hannes-mk** [comment]
> @hannes-mk I realized we could leave the date parsing to Spring by using `Instant` as the parameter type (rather than `String` + parse that ourselves). If we don't need to attempt the parsing ourselves we can simplify the controller a lot. Only downside is that we can't have a 400 status code then. I think that's fine but would be curious to hear your feedback.

You mean the HTML path parameter could be `Instant` and then the endpoint would accept ISO, but reject invalid ISO?
Works for me. Let's have a quick chat so I learn how it would behave.

File: backend/src/main/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/controller/ElementController.java [hunk 1]
```diff
@@ -104,29 +103,21 @@ public ResponseEntity<String> getElementHtmlPreview(
...
-          .loadElementHtmlAtDateFromNorm(
-              new LoadElementHtmlAtDateFromNormUseCase.Query(
-                  eli, eid, Instant.parse(atIsoDate.get())))
-          .map(ResponseEntity::ok)
-          .orElse(ResponseEntity.notFound().build());
-    }
-
-    return loadElementHtmlFromNormUseCase
-        .loadElementHtmlFromNorm(new LoadElementHtmlFromNormUseCase.Query(eli, eid))
+    return atIsoDate
```
Comment (hannes-mk) [inline backend/src/main/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/controller/ElementController.java:112]: I'm not sure I like this construct' use of `map`. 
Let's compare the alternatives.

**Review by hannes-mk — APPROVED** [review]
Nothing wrong, but I'd like to learn more about the alternatives.

**Pull request merged**